charts/influxdb/templates/backup-cronjob.yaml

```diff
@@ -2,102 +2,101 @@
-  schedule: {{ .Values.backup.schedule | quote }}
+  schedule: {{.Values.backup.cronjob.schedule | quote }}
   concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: {{ .Values.backup.cronjob.historyLimit }}
+  failedJobsHistoryLimit: {{ .Values.backup.cronjob.historyLimit }}
   jobTemplate:
     spec:
       template:
         metadata:
-          {{- if .Values.backup.podAnnotations }}
-          annotations:
-            {{ toYaml .Values.backup.podAnnotations | nindent 12 }}
-          {{- end }}
           labels:
             {{- include "influxdb.selectorLabels" . | nindent 12 }}
+          annotations:
+            {{- toYaml .Values.backup.cronjob.Podannotations | nindent 12 }}
         spec:
           restartPolicy: OnFailure
           volumes:
-          - name: backup
-            emptyDir: {}
-          {{- if .Values.backup.gcs }}
-          {{- if .Values.backup.gcs.serviceAccountSecret }}
-          - name: google-cloud-key
-            secret:
-              secretName: {{ .Values.backup.gcs.serviceAccountSecret | quote }}
-          {{- end }}
-          {{- end }}
-          serviceAccountName: {{ include "influxdb.serviceAccountName" . }}
+            - name: backup-scripts
+              configMap:
+                name: {{ include "influxdb.fullname" . }}-backup
+                defaultMode: 0755
+            {{- if .Values.backup.uploadProviders.google.enabled }}
+            - name: google-cloud-key
+              secret:
+            {{- if .Values.backup.uploadProviders.google.existingSecret }}
+                secretName: {{ .Values.backup.uploadProviders.google.existingSecret | quote }}
+            {{- else }}
+                secretName: {{ include "influxdb.fullname" . }}-backup-gcloud
+            {{- end }}
+            {{- end }}
+            - name: {{ include "influxdb.fullname" . }}-backups
+            {{- if .Values.persistence.enabled }}
+              persistentVolumeClaim:
+                claimName: {{ include "influxdb.fullname" . }}-backups
+            {{- else }}
+                emptyDir: {}
+            {{- end }}
           initContainers:
-          - name: influxdb-backup
-            image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-            volumeMounts:
-            - name: backup
-              mountPath: /backup
-            command:
-            - /bin/sh
-            args:
-            - '-c'
-            - |
-              influxd backup -host {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc:{{ .Values.config.rpc.bind_address | default 8088 }} -portable /backup/$(date +%Y%m%d%H%M%S)
+            - name: influxdb-backup
+              image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+              env:
+                - name: INFLUXDB_ADMIN_USER_PASSWORD
+                  valueFrom:
+                    secretKeyRef:
+                      name: {{ include "influxdb.fullname" . }}-auth
+                      key: influxdb-password
```

Comment:
Please wrap around if statement, auth may not be set and it will look for a non-existing key.

Reply:
Without this the backup script won't work as it queries the existing databases via this user.
I've added a note to the values file that you need the defaultuser.

Reply:
What happens with those queries if the user does not exist and the dbs are not user/pass protected? If it breaks define the args based on if 

While it is not a best practice, it is surely possible to have passwordless dbs and those users should be able to backup their data. https://github.com/influxdata/helm-charts/blob/e020a4b449041c31449d867e12e772e0224f9dd4/charts/influxdb/values.yaml#L99-L100 as this is set by default, backups should adhere to that as well.